run-docker-sample.sh

```diff
@@ -18,26 +18,51 @@
 #
 # ------------------------------------------------------------------------
 
+service_provider_name=tomcat-saml-agent-sample
+
+echo "1. Starting Asgardio Tomcat SAML Agent Sample container: "
 docker container run --rm --name tomcat-sample -itdp 8080:8080 tomcat-saml-agent
 
+echo "2. Starting WSO2 Identiy Server container: "
 docker container run --rm -itdp 9443:9443 --link tomcat-sample wso2/wso2is
 
-sleep 100s
+while ! curl -sk https://localhost:9443/api/health-check/v1.0/health > /dev/null
+do
+  sleep 5s
+  echo "Waiting for the WSO2 Identity Server to start..."
+done
 
-curl -kv --location --request POST 'https://localhost:9443/services/IdentityApplicationManagementService' \
+echo "3. Creating Service Provider: $service_provider_name"
+curl -sk --location --request POST 'https://localhost:9443/services/IdentityApplicationManagementService' \
 --header 'SOAPAction: createApplication' \
 --header 'Content-Type: text/xml' \
 --header 'Authorization: Basic YWRtaW46YWRtaW4=' \
---data-raw '<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsd="http://org.apache.axis2/xsd" xmlns:xsd1="http://model.common.application.identity.carbon.wso2.org/xsd" xmlns:xsd2="http://script.model.common.application.identity.carbon.wso2.org/xsd">    <soapenv:Header/>    <soapenv:Body>       <xsd:createApplication>          <xsd:serviceProvider>             <xsd1:applicationName>tomcat-saml-agent-sample</xsd1:applicationName>             <xsd1:inboundAuthenticationConfig>             <xsd1:inboundAuthenticationRequestConfigs>               <xsd1:inboundAuthKey>sample-app</xsd1:inboundAuthKey>               <xsd1:inboundAuthType>samlsso</xsd1:inboundAuthType>             </xsd1:inboundAuthenticationRequestConfigs>             </xsd1:inboundAuthenticationConfig>          </xsd:serviceProvider>       </xsd:createApplication>    </soapenv:Body> </soapenv:Envelope>'
+--data-raw '<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsd="http://org.apache.axis2/xsd" xmlns:xsd1="http://model.common.application.identity.carbon.wso2.org/xsd" xmlns:xsd2="http://script.model.common.application.identity.carbon.wso2.org/xsd">    <soapenv:Header/>    <soapenv:Body>       <xsd:createApplication>          <xsd:serviceProvider>             <xsd1:applicationName>'"$service_provider_name"'</xsd1:applicationName>             <xsd1:inboundAuthenticationConfig>             <xsd1:inboundAuthenticationRequestConfigs>               <xsd1:inboundAuthKey>sample-app</xsd1:inboundAuthKey>               <xsd1:inboundAuthType>samlsso</xsd1:inboundAuthType>             </xsd1:inboundAuthenticationRequestConfigs>             </xsd1:inboundAuthenticationConfig>          </xsd:serviceProvider>       </xsd:createApplication>    </soapenv:Body> </soapenv:Envelope>' \
```

Comment:
Extract the soap message to a separate variable.

Reply:
Resolved in https://github.com/asgardio/asgardio-tomcat-saml-agent/pull/13/commits/d14f65689a5380fe93f51e02ff4e1c77e255c505